### ISSUE TITLE
BERT training using MirroredStrategy

### ISSUE THREAD

**Issue by kamalkraj** [tensorflow/models#8433]
While going through the `run_customized_training_loop` function in BERT which is used for model training using multi-GPU I can't find the usage of `tf.distribute.Strategy.reduce`. How is the loss across multiple replicas are aggregated? @jharmsen @lehougoogle @chenGitHuber @saberkun 

https://www.tensorflow.org/guide/distributed_training#using_tfdistributestrategy_with_custom_training_loops

**Comment by saberkun** [comment]
@kamalkraj Hi, thanks for the question. I believe this is a very common question.
Current tf.keras.optimizers.Optimizer does the all reduce for you when it applies gradients. 
See: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L465
There is one special arg added recently: experimental_aggregate_gradients, which is to disable this behavior. If it is disabled, you need to explicitly reduce gradients in customized training loop.
We get this implicit reduce issue for readability and flexibility to TF/Keras team and the team is working on a new design for optimizer interface.

**Comment by kamalkraj** [comment]
@saberkun Thank you
